### ISSUE TITLE
[Bug] Heimer suddenly crashing randomly since 2.2.0

### ISSUE THREAD

**Issue by cantalisa** [juzzlin/Heimer#146]
Hello.

I've tried Heimer for a while, but apparently I'm experiencing instability issues since version 2.2.0. It closes suddenly. It is almost random, but perhaps at the perception level I have noticed that it relates to handling or modifying nodes (sometimes when you hover over them, there are no options to move, add node or change color). But it could be another bug independent of this one.

I can't give much detail about it and I don't know how it's reproducible. Generally, I have noticed that it closes when trying to delete text from a node, or try to modify text. It just closes or when I'm going to write in another one or changing text from a node.

This error occurs in both Manjaro, Ubuntu and Windows. I am a common user and do not know technical aspects, but if I can provide more information I can gladly try to provide it.

Since Heimer does not have an automatic save option this causes much of my work to be lost continuously. Heimer can even be closed by simply pressing Ctrl + S when trying to save the current job. It reaches this level of volatility.

Cheers.

**Comment by juzzlin** [comment]
Thanks for reporting! 

Have you installed Heimer from the official packages?

**Comment by juzzlin** [comment]
Unfortunately I cannot make the version in current master crash - at least not on Ubuntu. I would appreciate some more detailed sequence to make this happen. These random crashes can be very difficult to deal with :( 

Also, the automatic save option is on my work list.

**Comment by cantalisa** [comment]
> Thanks for reporting!
> 
> Have you installed Heimer from the official packages?

Hi, thanks for responding so quickly!

Yes, I use the assets (.exe and .deb) from this same repository. Except maybe in Manjaro, where I believe in the AUR it seems outdated, but it belongs to this same repository.

When I write it does not crash, but sometimes it seems that when I press the Delete key when I am out of the writing area it seems to be that it breaks (perhaps due to some combination of keys). Since on one occasion it even crashed when I put Ctrl + S when trying to save.

I can over the course of this week record the screen and do tests at the time when Heimer closes. Since it seems random. I just type and drag to move continuously. Or maybe because of the number of nodes that I handle, which are usually relatively complex.

By the way, great to hear that the autosave feature is in development! I have recommended Heimer to several fellow students and some have also wondered about it. I find it fantastic.

Thank you very much for Heimer. I don't know anything about code, but if I can support the project in some way, let me know, with great pleasure.

Cheers.

**Comment by l3td33r** [comment]
I managed to crash it once, using 2.2.0 version. Segmentation fault error, no details.
No idea how it happened, I created ~100 nodes, optimized layout and then started clicking, dragging random nodes and using random keyboard buttons as fast as I could, until it crashed.

**Comment by l3td33r** [comment]
Not sure if it is related, because it doesn't happen on 2.2.0, but on the latest master https://github.com/juzzlin/Heimer/commit/52f0f987639d5d457ed97470c5edd8c1cc2e5ff6
there's confirmed crash.

Created an issue for it https://github.com/juzzlin/Heimer/issues/150

**Comment by juzzlin** [comment]
Optional autosave implemented in 3ff6514a834221602c6802219861fcd49ab94f9c.

**Comment by cantalisa** [comment]
Hello. An apology for the somewhat late response.

I've been testing during both last and this week with the maps that I considered complex and that Heimer used to crash to (considering also that it could crash into new or genuinely very simple maps). I have recorded screen but it seems not useful or necessary to go up, because on these occasions, Heimer did not close at any time.

The issue #150 created by @l3td33r seems to make sense and may be because of the same issue, since that day when I posted this issue I worked quickly and perhaps the issue between quickly switching between writing and drag and drop or moving continuously is what I perceived.

Should I close this Issue, or do I keep testing Heimer to crash again?

Cheers.
